### PR TITLE
Only raise exception if a list is None

### PR DESCRIPTION
**Pull request by adamtheturtle**
https://jira.mesosphere.com/browse/DCOS-20200

### CODE REVIEW

File: dcos_test_utils/dcos_api.py [hunk 1]
```diff
@@ -364,15 +364,18 @@ def wait_for_dcos(self):
...
+        if wait_for_hosts and not node_lists_set:
             raise Exception(
                 'This cluster is set to wait for hosts, however, not all host lists '
                 'were suppplied. Please set all three environment variables of MASTER_HOSTS, '
                 'SLAVE_HOSTS, and PUBLIC_SLAVE_HOSTS to the appropriate cluster IPs (comma separated). '
                 'Alternatively, set WAIT_FOR_HOSTS=false in the environment to use whichever hosts '
                 'are currently registered.')
-        else:
-            self.set_node_lists_if_unset()
+        self.set_node_lists_if_unset()
```
Comment (adamtheturtle) [inline dcos_test_utils/dcos_api.py]: Moving this out of an unnecessary `else` is unrelated.

**Comment by adamtheturtle** [comment]
I have not tested this.

**Comment by adamtheturtle** [comment]
@mellenburg, I reported this issue because I saw the following:

{code}
[01:01:43][Step 4/4]     def wait_for_dcos(self):
[01:01:43][Step 4/4]         self._wait_for_adminrouter_up()
[01:01:43][Step 4/4]         self._authenticate_default_user()
[01:01:43][Step 4/4]         wait_for_hosts = os.getenv('WAIT_FOR_HOSTS', 'true') == 'true'
[01:01:43][Step 4/4]         if wait_for_hosts and (not self.master_list or not self.slave_list or not self.public_slave_list):
[01:01:43][Step 4/4]             raise Exception(
[01:01:43][Step 4/4] >               'This cluster is set to wait for hosts, however, not all host lists '
[01:01:43][Step 4/4]                 'were suppplied. Please set all three environment variables of MASTER_HOSTS, '
[01:01:43][Step 4/4]                 'SLAVE_HOSTS, and PUBLIC_SLAVE_HOSTS to the appropriate cluster IPs (comma separated). '
[01:01:43][Step 4/4]                 'Alternatively, set WAIT_FOR_HOSTS=false in the environment to use whichever hosts '
[01:01:43][Step 4/4]                 'are currently registered.')
[01:01:43][Step 4/4] E           Exception: This cluster is set to wait for hosts, however, not all host lists were suppplied. Please set all three environment variables of MASTER_HOSTS, SLAVE_HOSTS, and PUBLIC_SLAVE_HOSTS to the appropriate cluster IPs (comma separated). Alternatively, set WAIT_FOR_HOSTS=false in the environment to use whichever hosts are currently registered.
[01:01:43][Step 4/4] 
{code}

on one of @takirala 's  builds, which was at https://teamcity.mesosphere.io/viewLog.html?buildId=927405&buildTypeId=DcOs_Enterprise_Components_PackageRegistry_Packag_PackageRegistryClusterIntegrat&tab=buildLog but now I seemingly cannot get to ("Build with id '927405' not found").

I'm not sure exactly how to recreate this.
Do you have a good testing method for this function?
To me, eyeballing this change, it seems sensible.
Please can you use your judgement, or let me know how you would like me to test this.

**Comment by mellenburg** [comment]
@adamtheturtle This makes sense to me and its pretty clear that the error snippet you posted above will occur with any test run that has zero agents of a given group :grimacing: So, this LGTM and we should get it merged ASAP to make zero-private-agent and/or zero-public-agent clusters testable again

**Comment by takirala** [comment]
I might have removed that build due to some restructuring. Thanks for the fix @adamtheturtle !

**Pull request merged**